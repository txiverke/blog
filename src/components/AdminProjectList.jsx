// @flow

import React from 'react'
import { connect } from 'react-redux'
import ReactMessages from 'react-messages'

import AdminList from '../containers/AdminList'
import AdminProjectItem from '../components/AdminProjectItem'
import Loader from './Loader'
import { loadProjectData } from '../actions/projectActionCreators'

class AdminProjectList extends React.Component {
  
  props: {
    dispatch: Function,
    projects: Data,
    handleCreateProject: Function
  }

  componentDidMount() {
    const { dispatch, projects } = this.props
    
    if (projects.data && projects.data.length === 0) {
      dispatch(loadProjectData())
    }
  }

  createPost = (obj: Object) => {
    this.props.handleCreateProject(obj)
  }

  render() {
    const { data, completed, message, error } = this.props.projects

    if (completed) {
      return (
        <div className="app-content-grid"> 
          <ReactMessages message={message} error={error} next={true} />
          <AdminProjectItem label="Create Project" handleProject={this.createPost} />
          <AdminList type="projects" data={data} />
        </div>
      )
    }

    return <Loader msg={message} />
  }
}

export default connect()(AdminProjectList)