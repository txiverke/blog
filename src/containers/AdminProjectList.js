// @flow

import { connect } from 'react-redux'

import AdminProjectList from '../components/AdminProjectList'
import { createProjectData } from '../actions/projectActionCreators'

const mapStateToProps = state => ({ projects: state.projects })

const mapDispatchToProps = (dispatch: Function) => ({
  handleCreateProject: (obj: Object) => dispatch(createProjectData(obj))
})

export default connect(mapStateToProps, mapDispatchToProps)(AdminProjectList)