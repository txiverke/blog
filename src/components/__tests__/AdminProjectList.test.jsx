// @flow

import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { Unwrapped as UnwrappedAdminProjectList } from '../AdminProjectList'

const properties = {
  dispatch: jest.fn(),
  projects: {
    completed: true,
    data: [],
    message: 'fake message',
    error: false
  },
  handleCreateProject: jest.fn()
}

describe('<AdminPostList />', () => {
  it('should render itself', () => {
    const component = shallow(<UnwrappedAdminProjectList {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should render Loader if there is not data', () => {
    properties.projects.completed = false
    const component = shallow(<UnwrappedAdminProjectList {...properties} />)
    expect(component.find('Loader').length).toEqual(1)  
  })

  it('should render the list if there is data', () => {
    properties.projects.completed = true
    const component = shallow(<UnwrappedAdminProjectList {...properties} />)
    expect(component.find('Loader').length).toEqual(0)
    expect(component.find('.app-content-grid').length).toEqual(1)      
  })

})