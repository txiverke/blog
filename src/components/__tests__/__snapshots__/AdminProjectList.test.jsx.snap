// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPostList /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminProjectList
    dispatch={[Function]}
    handleCreateProject={[Function]}
    projects={
        Object {
            "completed": true,
            "data": Array [],
            "error": false,
            "message": "fake message",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <M
          duration={2000}
          error={false}
          icon="heart"
          message="fake message"
          next={true}
/>,
        <AdminProjectItem
          handleProject={[Function]}
          label="Create Project"
/>,
        <Connect(AdminList)
          data={Array []}
          type="projects"
/>,
      ],
      "className": "app-content-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "duration": 2000,
          "error": false,
          "icon": "heart",
          "message": "fake message",
          "next": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleProject": [Function],
          "label": "Create Project",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [],
          "type": "projects",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <M
            duration={2000}
            error={false}
            icon="heart"
            message="fake message"
            next={true}
/>,
          <AdminProjectItem
            handleProject={[Function]}
            label="Create Project"
/>,
          <Connect(AdminList)
            data={Array []}
            type="projects"
/>,
        ],
        "className": "app-content-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "duration": 2000,
            "error": false,
            "icon": "heart",
            "message": "fake message",
            "next": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleProject": [Function],
            "label": "Create Project",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "type": "projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
